### ISSUE TITLE
Write the code

### ISSUE THREAD

**Issue by gobbedy** [gobbedy/deep_learning_adversarial_training#2]
1) Implement 2 simple classifiers MNIST and CIFAR classifiers
- Must use tensorflow, not pytorch
- For MNIST, can pick algorithm from assignment 2 (either MLP or CNN -- softmax is too simple)
- Look online for CIFAR, pretty sure there is pre-existing code

2) For each sample in a batch, find the closest adversarial example, call it XA
- Until Lit review is done, for now use the Carlini & Wagner "L2 attack", for which we have the code: https://github.com/carlini/nn_robust_attacks/blob/master/l2_attack.py
- Note that using Carlini method, to find the closest adversarial example, need to perform targeted attack for every adversarial class; then find the closest one

3) For each training sample in batch, once you have closest adversarial example, find nearest neighbor in original (or label) class -- call this XL, and nearest neighbor in target (or adversarial) class -- call this XT
- There are existing optimized algorithms for L2 nearest neighbor searches, we don't need to write it ourselves

4) Find the mean between XL and XT, call this XG (for generated). We hypothesize that XG is a good approximation of the true boundary
- This should be a one-liner

5) Now for each sample in the training batch, we have two samples -- the original and XG. The original sample gets trained as usual. XG is also passed into the classifier as usual, but the label is changed. Instead of a 1 for the original label class and zeros everywhere else, we assign 0.5 to the original label class, and 0.5 to the target (adversarial) class.
- I expect that this will be the most challenging modification to the original classifiers
